build/source/driver/multi_driver.f90

```diff
@@ -440,89 +615,90 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
+  fracHRU =  attrStruct%hru(iHRU)%var(iLookATTR%HRUarea) / bvarStruct%var(iLookBVAR%basin__totalArea)%dat(1)
 
-  ! assign pointers to model layers
+  ! assign model layers
   ! NOTE: layer structure is different for each HRU
-  nSnow   => indx_data%var(iLookINDEX%nSnow)%dat(1)
-  nSoil   => indx_data%var(iLookINDEX%nSoil)%dat(1)
-  nLayers => indx_data%var(iLookINDEX%nLayers)%dat(1)
+  nSnow(iHRU)   = indxStruct%hru(iHRU)%var(iLookINDEX%nSnow)%dat(1)
+  nSoil(iHRU)   = indxStruct%hru(iHRU)%var(iLookINDEX%nSoil)%dat(1)
+  nLayers       = indxStruct%hru(iHRU)%var(iLookINDEX%nLayers)%dat(1)
 
   ! get height at bottom of each soil layer, negative downwards (used in Noah MP)
-  allocate(zSoilReverseSign(nSoil),stat=err); call handle_err(err,'problem allocating space for zSoilReverseSign')
-  zSoilReverseSign(1:nSoil) = -mvar_data%var(iLookMVAR%iLayerHeight)%dat(nSnow+1:nSnow+nSoil)
-
-  ! assign pointers to model indices
-  midSnowStartIndex => indx_data%var(iLookINDEX%midSnowStartIndex)%dat(1)
-  midSoilStartIndex => indx_data%var(iLookINDEX%midSoilStartIndex)%dat(1)
-  midTotoStartIndex => indx_data%var(iLookINDEX%midTotoStartIndex)%dat(1)
-  ifcSnowStartIndex => indx_data%var(iLookINDEX%ifcSnowStartIndex)%dat(1)
-  ifcSoilStartIndex => indx_data%var(iLookINDEX%ifcSoilStartIndex)%dat(1)
-  ifcTotoStartIndex => indx_data%var(iLookINDEX%ifcTotoStartIndex)%dat(1)
+  allocate(zSoilReverseSign(nSoil(iHRU)),stat=err); call handle_err(err,'problem allocating space for zSoilReverseSign')
+  zSoilReverseSign(:) = -progStruct%hru(iHRU)%var(iLookPROG%iLayerHeight)%dat(nSnow(iHRU)+1:nLayers)
 
   ! get NOAH-MP parameters
-  call REDPRM(type_data%var(iLookTYPE%vegTypeIndex),                           & ! vegetation type index
-              type_data%var(iLookTYPE%soilTypeIndex),                          & ! soil type
-              type_data%var(iLookTYPE%slopeTypeIndex),                         & ! slope type index
+  call REDPRM(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex),                & ! vegetation type index
+              typeStruct%hru(iHRU)%var(iLookTYPE%soilTypeIndex),               & ! soil type
+              typeStruct%hru(iHRU)%var(iLookTYPE%slopeTypeIndex),              & ! slope type index
               zSoilReverseSign,                                                & ! * not used: height at bottom of each layer [NOTE: negative] (m)
-              nSoil,                                                           & ! number of soil layers
+              nSoil(iHRU),                                                     & ! number of soil layers
               urbanVegCategory)                                                  ! vegetation category for urban areas
 
   ! overwrite the minimum resistance
-  if(overwriteRSMIN) RSMIN = mpar_data%var(iLookPARAM%minStomatalResistance)
+  if(overwriteRSMIN) RSMIN = mparStruct%hru(iHRU)%var(iLookPARAM%minStomatalResistance)
```

Comment:
Vegetation height always comes from the veg tables. I wanted flexibility to do some runs with a specific minimum resistance. I also wanted to have backwards compatibility, hence the overwrite flag.

Ultimately this should be encoded as a specific modeling decision, though this is "on hold" while we work to get the spatial parameter fields organized.